setup.py

```diff
@@ -1,167 +1,64 @@
-
-npm_path = os.pathsep.join([
-    pjoin(node_root, 'node_modules', '.bin'),
-    os.environ.get('PATH', os.defpath),
-])
+long_description = 'Matplotlib Jupyter Extension'
 
 log.info('setup.py entered')
 log.info('$PATH=%s' % os.environ['PATH'])
 
-LONG_DESCRIPTION = 'Matplotlib Jupyter Extension'
-
-
-def js_prerelease(command, strict=False):
-    """decorator for building minified js/css prior to another command"""
-    class DecoratedCommand(command):
-        def run(self):
-            jsdeps = self.distribution.get_command_obj('jsdeps')
-            if not is_repo and all(os.path.exists(t) for t in jsdeps.targets):
-                # sdist, nothing to do
-                command.run(self)
-                return
-
-            try:
-                self.distribution.run_command('jsdeps')
-            except Exception as e:
-                missing = [t for t in jsdeps.targets if not os.path.exists(t)]
-                if strict or missing:
-                    log.warn('rebuilding js and css failed')
-                    if missing:
-                        log.error('missing files: %s' % missing)
-                    raise e
-                else:
-                    log.warn('rebuilding js and css failed (not a problem)')
-                    log.warn(str(e))
-            command.run(self)
-            update_package_data(self.distribution)
-    return DecoratedCommand
-
-
-def update_package_data(distribution):
-    """update package_data to catch changes during setup"""
-    build_py = distribution.get_command_obj('build_py')
-    distribution.data_files = get_data_files()
-    # re-init build_py options which load package_data
-    build_py.finalize_options()
-
-
-def get_data_files():
-    """Get the data files for the package.
-    """
-    with open(pjoin(node_root, 'package.json')) as f:
-        package_json = json.load(f)
-    tgz = '%s-%s.tgz' % (package_json['name'], package_json['version'])
-
-    return [
-        ('share/jupyter/nbextensions/jupyter-matplotlib', [
-            'ipympl/static/extension.js',
-            'ipympl/static/index.js',
-            'ipympl/static/index.js.map',
-            'ipympl/static/package.json'
-        ]),
-        ('etc/jupyter/nbconfig/notebook.d', ['jupyter-matplotlib.json']),
-        ('share/jupyter/lab/extensions', ['js/' + tgz]),
-    ]
-
+# Get ipympl version
+version = get_version(os.path.join(name, '_version.py'))
 
-class NPM(Command):
-    description = 'install package.json dependencies using npm'
+js_dir = os.path.join(here, 'js')
 
-    user_options = []
+# Representative files that should exist after a successful build
+jstargets = [
+    os.path.join(js_dir, 'dist', 'index.js'),
+]
 
-    node_modules = pjoin(node_root, 'node_modules')
+data_files_spec = [
+    ('share/jupyter/nbextensions/jupyter-matplotlib',
+     'ipympl/nbextension', '*.*'),
+    ('share/jupyter/labextensions/jupyter-matplotlib',
+     'ipympl/labextension', "*.*"),
```

Comment:
Note to @martinRenou the first two changes below seem to be necessary to make the resulting wheel file work.

```suggestion
    ('share/jupyter/nbextensions/jupyter-matplotlib',
     'ipympl/nbextension', '**'),
    ('share/jupyter/labextensions/jupyter-matplotlib',
     'ipympl/labextension', "**"),
    ('share/jupyter/labextensions/jupyter-matplotlib', '.', 'install.json'),
```
Based on https://github.com/jupyterlab/extension-cookiecutter-ts/blob/50a8b7eaa27af196439ba21cefff3d8237656776/%7B%7Bcookiecutter.python_name%7D%7D/setup.py#L37

Also I added the install.json file (https://github.com/jupyterlab/extension-cookiecutter-ts/blob/master/%7B%7Bcookiecutter.python_name%7D%7D/install.json) which will also need to be created